code/datums/chatmessage.dm

```diff
@@ -125,6 +125,39 @@
 		target.chat_color_darkened = colorize_string(target.name, 0.85, 0.85)
 		target.chat_color_name = target.name
 
+	// SKYRAT ADDITION - Character preference for chatmessages
+	if(ismob(target))
+		var/mob/speaker = target
+		if(speaker.client?.prefs)
+			// Apply the custom color if they want one
+			var/enable_chat_color_player = speaker.client.prefs.read_preference(/datum/preference/toggle/enable_chat_color_player)
+			if(enable_chat_color_player)
+				var/chat_color_player = speaker.client.prefs.read_preference(/datum/preference/color/chat_color_player)
+				var/hex_value = sanitize_hexcolor(chat_color_player)
+				target.chat_color = hex_value
+				target.chat_color_darkened = (hex_value + "c0") // Add an alpha-channel to darken
+
+			// If they are acting as someone else, find out who
+			if((speaker.name != speaker.real_name) && (speaker.name != "Unknown"))
+				for(var/mob/living/carbon/human/copy in GLOB.player_list)
```

Comment:
This should probably be an `as anything` to reduce the performance impact.

Also, what if they're using a name that's not theirs and there's nobody else with that name?

Reply:
I've thought about that, but not tested it. Like when you pick up a clown ID from a corpse on lavaland.
What I assume happens from reading the code is it doesn't continue to an if condition after the for loop, and it carries on, keeping the name-generated color from the original block above.
Will make sure this is the case, though.